### PR TITLE
Feature 18 Mechanism to open and use browser from application

### PR DESCRIPTION
**Pull request by vasylmrnchk**

### CODE REVIEW

File: RecodoDesktop/Recodo.Desktop.Logic/DefaultBrowser.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            }
+            catch (TaskCanceledException ex)
+            {
+                return ex.Message;
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/DefaultBrowser.cs:42]: Can we set a time to leave for this service? Or what will happen if user opens page, but will make nothing. And repeat this action 10 times? Will we listen to 10 ports?

File: RecodoDesktop/RecodoDesktop/AuthorizationWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public partial class AuthorizationWindow : Window
+    {
+        public AuthorizationWindow()
+        {
+            InitializeComponent();
+        }
+
+        private async void SignInButton_Click(object sender, RoutedEventArgs e)
+        {
+            await GetToken("http://127.0.0.1:4200/login");
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/RecodoDesktop/AuthorizationWindow.xaml.cs]: Move to some sort of settings, we won't have 127.0.0.1 always here

File: frontend/src/app/modules/auth/register-form/register-form.component.ts [hunk 3]
```diff
@@ -70,7 +78,13 @@ export class RegisterFormComponent {
     this.registrationService.register(userRegisterDto).subscribe((responce) => {
       this.currentUser = responce;
       if(this.registrationService.areTokensExist()) {
-        this.router.navigate(['/login']);
+        if (this.redirectUrl)
+        {
+          window.location.href= `${this.redirectUrl}?access_token=${localStorage.getItem('accessToken')}`;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/register-form/register-form.component.ts]: You can make it also as this.router.navigate

File: frontend/src/app/modules/auth/login-page/login-page.component.ts [hunk 4]
```diff
@@ -53,7 +58,13 @@ export class LoginPageComponent implements OnInit {
     this.loginService.login(_user).subscribe((responce) => {
       this.currentUser = responce;
       if(this.loginService.areTokensExist()) {
-        this.router.navigate(['/personal']);
+        if (this.redirectUrl)
+        {
+          window.location.href= `${this.redirectUrl}?access_token=${localStorage.getItem('accessToken')}`;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/auth/login-page/login-page.component.ts]: The same

File: RecodoDesktop/Recodo.Desktop.Logic/DefaultBrowser.cs (hunk 1 above)
Comment (vasylmrnchk) [reply RecodoDesktop/Recodo.Desktop.Logic/DefaultBrowser.cs:42]: Yes it's open ten ports, and if user nothing to do listener wait when expired timeout. After timeout expired it throw exception. Timeout set in the listener.

File: RecodoDesktop/Recodo.Desktop.Logic/AuthorizationService.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace Recodo.Desktop.Logic
+{
+    public class AuthorizationService
+    {
+        readonly private DefaultBrowser _browser;
+        readonly private string _redirect_url = $"http://{IPAddress.Loopback}:{GetRandomUnusedPort()}/";
+        readonly Dictionary<string, string> authRequestParams = new();
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/AuthorizationService.cs]: Move assignment to constructor

File: RecodoDesktop/Recodo.Desktop.Logic/LoopbackHttpListener.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Recodo.Desktop.Logic
+{
+    public sealed class LoopbackHttpListener: IDisposable
+    {
+        const int DefaultTimeout = 60 * 3;
+        readonly IWebHost _host;
+        readonly TaskCompletionSource<string> _source = new();
```
Comment (Anatolii-Ivanov) [inline RecodoDesktop/Recodo.Desktop.Logic/LoopbackHttpListener.cs]: The same

**Review by Jyuart — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**